backend/src/apiserver/server/pipeline_upload_server_test.go

```diff
@@ -459,3 +461,95 @@ func TestUploadPipelineVersion_FileNameTooLong(t *testing.T) {
+
+	clientManager := resource.NewFakeClientManagerOrFatal(util.NewFakeTimeForEpoch())
+	resourceManager := resource.NewResourceManager(clientManager)
+	server := PipelineUploadServer{resourceManager: resourceManager, options: &PipelineUploadServerOptions{CollectMetrics: false}}
+	b := &bytes.Buffer{}
+	w := multipart.NewWriter(b)
+	part, _ := w.CreateFormFile("uploadfile", "arguments.tar.gz")
+	fileReader, _ := os.Open("test/arguments_tarball/arguments.tar.gz")
+	io.Copy(part, fileReader)
+	w.Close()
+	req, _ := http.NewRequest("POST", "/apis/v1beta1/pipelines/upload", bytes.NewReader(b.Bytes()))
+	req.Header.Set("Content-Type", w.FormDataContentType())
+
+	rr := httptest.NewRecorder()
+	handler := http.HandlerFunc(server.UploadPipeline)
+	handler.ServeHTTP(rr, req)
+
+	pipelineVersion, err := clientManager.PipelineStore().GetPipelineVersion(resource.DefaultFakeUUID)
+	assert.Nil(t, err)
+	assert.Equal(t, pipelineVersion.PipelineId, resource.DefaultFakeUUID)
+
+	// Upload a new version under this pipeline and check that the default version is not updated
+
+	// Set the fake uuid generator with a new uuid to avoid generate a same uuid as above.
+	clientManager.UpdateUUID(util.NewFakeUUIDGeneratorOrFatal(fakeVersionUUID, nil))
+	resourceManager = resource.NewResourceManager(clientManager)
+	server = PipelineUploadServer{resourceManager: resourceManager, options: &PipelineUploadServerOptions{CollectMetrics: false}}
+	b = &bytes.Buffer{}
+	w = multipart.NewWriter(b)
+	part, _ = w.CreateFormFile("uploadfile", "arguments-version.tar.gz")
+	fileReader, _ = os.Open("test/arguments_tarball/arguments-version.tar.gz")
+	io.Copy(part, fileReader)
+	w.Close()
+	req, _ = http.NewRequest("POST", "/apis/v1beta1/pipelines/upload_version?pipelineid="+resource.DefaultFakeUUID, bytes.NewReader(b.Bytes()))
+	req.Header.Set("Content-Type", w.FormDataContentType())
+	rr = httptest.NewRecorder()
+	handler = http.HandlerFunc(server.UploadPipelineVersion)
+	handler.ServeHTTP(rr, req)
+
+	pipeline, err := clientManager.PipelineStore().GetPipeline(resource.DefaultFakeUUID)
+	assert.Nil(t, err)
+	assert.Equal(t, pipeline.DefaultVersionId, resource.DefaultFakeUUID)
+	assert.NotEqual(t, pipeline.DefaultVersionId, fakeVersionUUID)
+}
+
+func TestDefaultUpdatedPipelineVersion(t *testing.T) {
+	clientManager := resource.NewFakeClientManagerOrFatal(util.NewFakeTimeForEpoch())
+	resourceManager := resource.NewResourceManager(clientManager)
+	server := PipelineUploadServer{resourceManager: resourceManager, options: &PipelineUploadServerOptions{CollectMetrics: false}}
+	b := &bytes.Buffer{}
+	w := multipart.NewWriter(b)
+	part, _ := w.CreateFormFile("uploadfile", "arguments.tar.gz")
+	fileReader, _ := os.Open("test/arguments_tarball/arguments.tar.gz")
+	io.Copy(part, fileReader)
+	w.Close()
+	req, _ := http.NewRequest("POST", "/apis/v1beta1/pipelines/upload", bytes.NewReader(b.Bytes()))
+	req.Header.Set("Content-Type", w.FormDataContentType())
```

Comment:
nit: can we wrap this complex upload steps into a reusable function?
That helps making the test case easier to read.

Reply:
I agree, but maybe it is better to fix that in a separate PR since we should fix all tests in ``` pipeline_upload_server_test```

Reply:
Sure, logged https://github.com/kubeflow/pipelines/issues/4630